xtevent/_eventivstatic.ado

```diff
@@ -1,172 +1,184 @@
+	nofe /* No fixed effects */
+	note /* No time effects */	
+	reghdfe /* Use reghdfe for estimation */
+	absorb(string) /* Absorb additional variables in reghdfe */ 
+	impute(string)
+	STatic
+	*
+	]
+	;
+	#d cr
+	
+	marksample touse
+	
+	tempvar kg
+	* kg grouped event time, grouping outside window
+	
+	tempname delta Vdelta bb VV bb2 VV2 delta2 Vdelta2 deltay Vdeltay deltax Vdeltax deltaxsc bby bbx VVy VVx 
+	* bb delta coefficients
+	* VV variance of delta coefficients
+	* bb2 delta coefficients for overlay plot
+	* VV2 variance of delta coefficients for overlay plot
+	* delta2 included cefficientes in overlaty plot
+	* VVdelta2 variance of included delta coefficients in overlay plot
+	
+	loc i = "`panelvar'"
+	loc t = "`timevar'"
+	loc z = "`policyvar'"
+	
+	loc leads : word count `proxy'
+	if "`proxyiv'"=="" & `leads'==1 loc proxyiv "select"
+	
+	* If proxy specified but no proxyiv, assume numlist for leads of policyvar
+	if "`proxyiv'"=="" {
+		di _n "No proxy instruments specified. Using leads of policy variables as instruments."
+		loc leads : word count `proxy'		
+		forv j=1(1)`leads' {
+			loc proxyiv "`proxyiv' `j'"
+		}
+	}
+	
+	* IV selection if proxyiv = selection
+	else if "`proxyiv'"=="select" {
+		* Only for one proxy case 		
+		if `leads'>1 {
+			di as err "Proxy instrument selection only available for the one proxy - one instrument case"
+			exit 301
+		}
+		else {
+			di as text _n "proxyiv=select. Selecting lead order of differenced policy variable to use as instrument."
+			loc Fstart = 0
+			* Here I test up to 5
+			forv v=1(1)5 {
+				tempvar _fd`v'`z'
+				qui gen `_fd`v'`z'' = f`v'.d.`z' if `touse'
+				cap qui reg `proxy' `_fd`v'`z'' [`weight'`exp'] if `touse'
+				if !_rc loc Floop = e(F)
+				if `Floop' > `Fstart' {
+					loc Fstart = `Floop'
+					loc proxyiv "`v'"				
+				}			
+			}
+			di as text _n "Lead `proxyiv' selected."
+		}
+		
+	}
+	
+	* Parse proxyiv and generate leads if neccesary
+	loc rc=0
+	foreach v in `proxyiv' {
+		cap confirm integer number `v'
+		loc rc = `rc' + _rc
+	}
+	* If numlist take as leads of z
+	if `rc' == 0 {
+		loc insvars ""
+		foreach v in `proxyiv' {
+			qui gen _f`v'`z' = f`v'.`z' if `touse'
+			loc insvars "`insvars' _f`v'z"
+		}
+		loc instype = "numlist"
+	}
+	else {
+		foreach v in `proxyiv' {
+			confirm numeric variable `v'
+		}
+		loc insvars = "`proxyiv'"
+	}
+	
+	*call _eventgenvars to impute z
+	_eventgenvars if `touse', panelvar(`panelvar') timevar(`timevar') policyvar(`policyvar') impute(`impute') `static'
+	
+		
+	* Main regression
+	
+	if "`impute'"!="" {
+		loc zreg="`policyvar'_imputed"
+	}
+	else {
+		loc zreg="`z'"
+	}
+	
+	if "`reghdfe'"=="" {
+		if "`fe'" == "nofe" {
+		loc cmd "ivregress 2sls"
+		loc ffe ""
+		}
+		else {
+			loc cmd "xtivreg"
+			loc ffe "fe"
+		}
+		`cmd' `varlist' (`proxy' = `insvars') `zreg' `tte' [`weight'`exp'] if `touse' , `ffe' `options'
+	}
+	else {

```

Comment:
Are these conditionals trying to capture all combinations of unit fixed effects, time fixed effects, and clustering options? If so, we had `2^3 = 8` conditionals in [eventstudyr](https://github.com/JMSLab/eventstudyr/blob/main/R/EventStudyOLS.R) and just wanted to make sure that all cases are being accounted for?

Reply:
thanks, I'm changing conditions to cover all possibilities

Reply:
@jorpppp the default option for eventstudyr is to cluster

Reply:
Thanks @nateschor. Maybe it would be a good idea not to cluster by default in eventstudyr to maintain consistency?
At the time when I was deciding the default for xtevent I crowdsourced it and people preferred no clustering by default:
https://twitter.com/jorpppp/status/1204166283819978752 
But maybe this preference was because it was more natural to have this default for Stata syntax. Do you think it's more natural to have a clustering default in R?

Reply:
@veli-m-andirin do you have any sense of whether R users would prefer clustering or no clustering to be the default? Also should we have Richard added to this repo?

Since we're using [estimatr::lm_robust](https://www.rdocumentation.org/packages/estimatr/versions/0.30.6/topics/lm_robust) for OLS and the default standard error argument is NULL and the user has to specify which variable(s) to cluster on if they would like clustering, changing the default to not cluster makes sense to me!

Reply:
@veli-m-andirin @nateschor  Since this is no longer an -xtevent- issue I'm going to mark this as resolved, but let me know what you decide about the default clustering options for -eventstudyR-